src/vtp/cast_ballot.py

```diff
@@ -19,316 +19,36 @@
-        # pylint: disable=broad-except
-        except Exception:
-            # blow out of the internal pyinputplus try/catch
-            traceback.print_exc()
-            sys.exit(1)
-
-    if tally == "plurality":
-        if max_votes > 1:
-            prompt = (
-                f"Please enter the numbers for your choices (max={max_votes}) "
-                "separated by spaces: "
-            )
-        else:
-            prompt = "Please enter the number for your choice: "
-        pyinputplus.inputCustom(validate_multichoice, prompt=prompt, blank=True)
-    else:
-        # Then prompt for input
-        prompt = "Please enter in rank order the numbers of your choices separated by spaces: "
-        pyinputplus.inputCustom(validate_multichoice, prompt=prompt, blank=True)
-
-
-def loop_over_contests(a_ballot):
-    """Will loop over the contests in a ballot and either ask the user
-    for a choice or if in demo mode will randomly choose one.
-    """
-    contests = Contests(a_ballot)
-    total_contests = contests.len()
-    count = 0
-    contest_uids = []
-    for contest in contests:
-        contest_uids.append(contest.get("uid"))
-        if ARGS.demo_mode:
-            make_random_selection(a_ballot, contest)
-        else:
-            # Display the tally type and choices and allow the user to manually
-            # enter something.  Might as well validate legal selections (in
-            # this demo) as that is the long-term VTP vision.
-            count += 1
-            get_user_selection(a_ballot, contest, count, total_contests)
-    if not ARGS.demo_mode:
-        # UX wise replicate the self adjudication experince.  This is
-        # basically another endless loop until done
-        while True:
-            # Print the selections
-            for contest in contests:
-                print(
-                    f"Contest {contest.get('uid')} - {contest.get('name')}: "
-                    f"{contest.get('selection')}"
-                )
-            prompt = "Is this correct?  Enter yes to accept the ballot, no to reject the ballot: "
-            if "yes" == pyinputplus.inputYesNo(prompt):
-                break
-            prompt = (
-                "Enter a contest uid to redo that contest, "
-                "enter nothing (leave blank and hit enter) to start completely over: "
-            )
-            response = pyinputplus.inputChoice(contest_uids + [""], prompt)
-            if response == "":
-                count = 0
-                for contest in contests:
-                    count += 1
-                    get_user_selection(a_ballot, contest, count, total_contests)
-            else:
-                for contest in contests:
-                    if contest.get("uid") == response:
-                        get_user_selection(a_ballot, contest, 1, 1)
-                        break
-    # For a convenient side effect, return the contests
-    return contests
-
-
```

Comment:
Keep everything below this point in this file. This is the script, argument parsing belongs here, separate from the internal operations.

Reply:
Marking this resolved as it is covered by the preceding conversation.